configuration/exodus-pipeline.yaml

```diff
@@ -273,15 +283,16 @@ Resources:
   CodePipelineCloudTrail:
     Type: AWS::CloudTrail::Trail
     # Create trail for prod pipeline
-    Condition: IsProd
+    Condition: IsExodusProd
     Properties:
-      S3BucketName: exodus-pipeline-artifacts
+      S3BucketName: cloudtrail-logs
       EventSelectors:
         - DataResources:
           - Type: AWS::S3::Object
             Values:
-              - arn:aws:s3:::exodus-pipeline-artifacts/build-stage
+              - "arn:aws:s3:::"
           ReadWriteType: WriteOnly
+          IncludeManagementEvents: false
       IncludeGlobalServiceEvents: true
       IsLogging: true
       IsMultiRegionTrail: true
```

Comment:
Monitoring all s3 actions across the account seems like it would get very expensive very fast (log storage in s3).
I don't think it's really necessary either, as our only use for this right now is for pipeline activities. CloudTrail doesn't need to know about every change in the CDN buckets, for instance -- no action needs to be taken for those events.

According to lines 275-277, every "project" is using *-pipeline-artifacts/build-stage as the object for triggering its prod pipeline, so I'm not sure why the EventSelectors needs this change anyway. Can't it just be `arn:aws:s3:::${project}-pipeline-artifacts/build-stage`?

Reply:
> Monitoring all s3 actions across the account seems like it would get very expensive very fast (log storage in s3).
> I don't think it's really necessary either, as our only use for this right now is for pipeline activities. CloudTrail doesn't need to know about every change in the CDN buckets, for instance -- no action needs to be taken for those events.

If we decide to use a separate AWS account for the exodus dev env pipeline (as briefly discussed/suggested in one of our sync meetings a while ago), we could also use this account for our developer testing pipelines. I don't suppose we would have too many s3 bucket writes in our development account. If we have 5 million s3 object writes per month, storage costs alone would be around $0.50/mo. API calls to the s3 buckets for 5 million items would be $50/mo (deletes are free). CloudTrail pricing for 5 million s3 object writes would also be $50/mo. I don't know how many s3 write events we would do per month on a production account, but I would guess less than 100,000 on our development accounts per month. So, CloudTrail costs on our development env would would be quite low. I have also added a lifecycle rule to the cloudtrail logs bucket that will delete the logs after one day to further reduce costs. In such a case, we could use one cloudtrail for the developement aws account that monitors all writes to buckets, and one for the production account (that contains the stage and prod envs) that monitors only the exodus-pipeline-artifacts bucket. 

> According to lines 275-277, every "project" is using *-pipeline-artifacts/build-stage as the object for triggering its prod pipeline, so I'm not sure why the EventSelectors needs this change anyway. Can't it just be arn:aws:s3:::${project}-pipeline-artifacts/build-stage?

No, because using a variable name in the CloudTrail bucket definition will either create a new CloudTrail trail (in the event that a name is not specified for the resource) for the {project}-pipeline, or, if we assign the CloudTrail resource a name, and try to update it with arn:aws:s3:::${project}-pipeline-artifacts/build-stage for each project, CloudFormation will complain that the resource already exists in another stack, and will refuse to update the resource. There are other options. We could maintain a list of developers' buckets in our CloudFormation templates, or share one or two pipeline-artifacts buckets. We could also disable the automatic pipeline run on developers' prod pipelines. I've outlined these options in my comment in RHELDST-1035.

> Is there some other use case in which broader monitoring is necessary?

I do not have a specific idea in mind, but it might be useful to monitor API calls on our cdn buckets. It seems like it could be useful for security and/or auditing purposes.

Reply:
Ah, I hadn't considered a change to the event arn creating a new trail. So, yeah, I suppose if we want to use this for all pipelines we either need to monitor all buckets or force all the pipelines to use a single bucket, which probably wouldn't work out well.